### PR TITLE
Dvc 2951 add logging to config manager

### PR DESCRIPTION
**Pull request by emmawillis**

### CODE REVIEW

**Comment by emmawillis** [comment]
Not sure if i should include the Nlog directory in the commit or not - anyone know?

**Comment by squarepegg** [comment]
You'll probably be very annoyed with me :/

Had a realization that as this is a library, we should provide an ILogger interface which logs things instead of dictating to the sdk user that they must use NLog to use this SDK.

The sdk user can then add the provider to utilize ILogger inside their app.

**Comment by emmawillis** [comment]
All good! That makes more sense for sure, I'll fix!

File: DevCycle/ConfigManager/EnvironmentConfigManager.cs [hunk 1]
```diff
@@ -68,22 +71,27 @@ private void SetConfig(IRestResponse res)
         {
             if (res.StatusCode == HttpStatusCode.NotModified)
             {
-                //TODO log (`Config not modified, using cache, etag: ${ this.configEtag}`)
+                if (logger != null) logger.LogInformation("Config not modified, using cache, etag: " + configEtag);
```
Comment (squarepegg) [inline DevCycle/ConfigManager/EnvironmentConfigManager.cs:74]: I think you can still use something like 

```
logger.LogInformation($"Config not modified, using cache, etag: {configEtag");
```

File: DevCycle/DevCycle.DotNet.Server.SDK.csproj [hunk 2]
```diff
@@ -38,6 +38,9 @@
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
     <PackageReference Include="JsonSubTypes" Version="1.8.0" />
     <PackageReference Include="NETStandard.Library" Version="2.0.3" />
+    <PackageReference Include="NLog" Version="4.7.15" />
```
Comment (squarepegg) [inline DevCycle/DevCycle.DotNet.Server.SDK.csproj:41]: Should we remove NLog and only include it in the example project as an example of how to use the ILogger extension?

File: DevCycle/ConfigManager/EnvironmentConfigManager.cs (hunk 1 above)
Comment (emmawillis) [reply DevCycle/ConfigManager/EnvironmentConfigManager.cs]: Fixed, thanks!

File: DevCycle/DevCycle.DotNet.Server.SDK.csproj (hunk 2 above)
Comment (emmawillis) [reply DevCycle/DevCycle.DotNet.Server.SDK.csproj]: Good point, removed

**Review by squarepegg — APPROVED** [review]

**Pull request merged**